### DISCUSSION TITLE
Call different functions inside `queryFn` depending on condition?

### DISCUSSION THREAD

**Discussion by j-mattsson** [TanStack/query#6938] [Q&A] [answered]
Is it okay to call different functions inside queryFn depending on some condition?
Like this (calling either `x()` or `y()` depending on condition in `if`-statement):
```
useQuery(["abc", foo], () => {
    if (foo > 5) { /* or whatever... */
      return x(foo);
    }
    return y(foo);
  });
``` 

Or should queryFn always call one and the same function?

**Comment by TkDodo** [comment, marked as answer]
that's fine / react-query doesn't care. as long as `foo` is in the queryKey
